extensions/ql-vscode/src/databases.ts

```diff
@@ -680,6 +688,11 @@ export class DatabaseManager extends DisposableObject {
   private updatePersistedDatabaseList(): void {
     this.ctx.workspaceState.update(DB_LIST, this._databaseItems.map(item => item.getPersistedState()));
   }
+
+  private isExtensionControlledLocation(uri: vscode.Uri) {
+    const storagePath = this.ctx.storagePath || this.ctx.globalStoragePath;
+    return uri.path.startsWith(storagePath);
```

Comment:
This might need symlink resolution, e.g. call `realpath` on the two paths before checking the prefix.

Reply:
sure.